src/utl/src/ScopedTemporaryFile.cpp

```diff
@@ -33,4 +36,64 @@ ScopedTemporaryFile::~ScopedTemporaryFile()
   }
 }
 
+StreamHandler::StreamHandler(const char* filename, bool binary)
+    : filename_(filename)
+{
+  std::string tmp_filename = filename_ + ".tmp";
+  if (fs::exists(tmp_filename)) {
+    fs::remove(tmp_filename);
```

Comment:
hmmm... now I remember my concern. We really dont want the destructor to throw an exception, but just log errors. otherwise the system can leak memory when the rest of the destructor(member variables) does not run. I must admit that I dont know these C++ details on top of my head, but it is a mentioned concern with RAII.

Reply:
Also, when we finally do arrive at the exact implementation, we need some succinct comments why the code is exactly this way or ideally tests to enforce it...